storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -290,7 +320,15 @@ public Optional<String> serverConfigName(String configName) {
             .define(TopicConfig.LOCAL_LOG_RETENTION_MS_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_MS, atLeast(-2), MEDIUM,
                 TopicConfig.LOCAL_LOG_RETENTION_MS_DOC)
             .define(TopicConfig.LOCAL_LOG_RETENTION_BYTES_CONFIG, LONG, DEFAULT_LOCAL_RETENTION_BYTES, atLeast(-2), MEDIUM,
-                TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC);
+                TopicConfig.LOCAL_LOG_RETENTION_BYTES_DOC)
+            .define(LOG_DIR_PROP, STRING, DEFAULT_LOG_DIR, HIGH, LOG_DIR_DOC)
+            .define(LOG_DIRS_PROP, STRING, null, HIGH, LOG_DIRS_DOC)
+            .define(METADATA_LOG_DIR_PROP, STRING, null, HIGH, METADATA_LOG_DIR_DOC)
+            .define(INTER_BROKER_PROTOCOL_VERSION_PROP, STRING, DEFAULT_INTER_BROKER_PROTOCOL_VERSION, new MetadataVersionValidator(), MEDIUM, INTER_BROKER_PROTOCOL_VERSION_DOC)
+            // This indicates whether unreleased APIs should be advertised by this node.
+            .defineInternal(UNSTABLE_API_VERSIONS_ENABLE_PROP, BOOLEAN, DEFAULT_UNSTABLE_API_VERSIONS_ENABLE, HIGH)
+            // This indicates whether unreleased MetadataVersions should be enabled on this node.
+            .defineInternal(UNSTABLE_METADATA_VERSIONS_ENABLE_PROP, BOOLEAN, DEFAULT_UNSTABLE_METADATA_VERSIONS_ENABLE, HIGH);
```

Comment:
I didn't see where we remove the definition from original kafkaConfig?

Reply:
KafkaConfig:446-447

Reply:
I mean, I can still see KafkaConfig:1305-1307 and other lines defined these configs, too.

Reply:
Right, missed that. Thanks.